src/infra/team-maintenance.md

```diff
@@ -4,34 +4,64 @@ The roster of the Rust teams is always in flux. From time to time, new people
 are added, but also people sometimes opt to into "alumni status", meaning that
 they are not currently an active part of the decision-making process.
 Unfortunately, whenever a new person is added or someone goes into alumni
-status, there are a number of disparate places that need to be updated. This
-page aims to document that list. If you have any questions, or need someone with
-more privileges to make a change for you, a good place to ask is `#infra` on
-Discord.
+status, there are a number of disparate places that need to be updated.
 
-### Team repo
+# Team repo
 
 Membership of teams is primarily driven by the config files in the
-[rust-lang/team repo][team repo]. Several systems use the team repo data to
-control access:
+[rust-lang/team repo][team repo]. See the README of that respository for the
+systems integrated with it.
 
-- the [team website]
-- bors r+ rights
-- rfcbot interaction
-- Mailgun email lists
+# Rules for changes to team repo
 
-Team membership is duplicated in a few other places listed below, but the
-long-term goal is to centralize on the team repo.
+Pull requests to the repository are merged by the [team-admins], who use these rules to merge PRs:
+
+### `people`, `teams`, and `repos` directories
+
+If a change is related to an individual and does not expand permissions, then only the individual's approval is required. If the change has already been made outside of the team repo (e.g., GitHub username change) then it is considered implicitly approved. This non-exhaustively includes:
+
+* Changing team membership to alumni or full removal
+* Changing email address
+* Adding zulip-id
+
+If a change will grant additional permissions, then a team lead needs to
+approve the change. Any team lead in the "parent team" chain may do so. This includes:
+
+* Adding new subteams under an existing team
+* Granting additional access (bors, dev-desktop access, etc.)
+* Changing other metadata (website descriptions, Zulip groups, etc.)
+* Changing access to repositories owned by their team
+  * For repositories ownership is not currently formally tracked. Until that is
+    added, the team-admins are expected to exercise their understanding of
+    which team owns the repository, when in doubt asking for clarification and
+    codifying in a comment in the relevant repository.
+
+### Source code changes
+
+The team repository additionally contains code to transform and validate the
+TOML user-edited files. This is owned by the Infrastructure team and approval
+should be sought for changes.
+
+### Who belongs to team-admins?
```

Comment:
Nit: `team-admins` might not be the best name, but I don't have a great alternative. Perhaps the more verbose `team-repo-admins`?